conans/client/conf/detect.py

```diff
@@ -1,16 +1,21 @@
 import os
 import platform
 import re
+import sys
 from subprocess import PIPE, Popen, STDOUT
 
 from conans.client.output import Color
 from conans.client.tools import detected_os, OSInfo
 from conans.client.tools.win import latest_visual_studio_version_installed
 from conans.model.version import Version
 
+isPY38 = bool(sys.version_info.major == 3 and sys.version_info.minor == 8)
+
 
 def _execute(command):
-    proc = Popen(command, shell=True, bufsize=1, stdout=PIPE, stderr=STDOUT)
+    # In Python 3.8, open() emits RuntimeWarning if buffering=1 for binary mode.
+    bufsize = -1 if isPY38 else 1
```

Comment:
The solution adopted in develop is:

```python
proc = subprocess.Popen(command, shell=True, bufsize=1, universal_newlines=True,
                            stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Maybe use it instead of bufsize?

Reply:
Oh, I didn't see the above. No problem.

Reply:
I saw it in `develop`, but it is the same line with a `bufsize=1` and I don't know why that one is not failing... so it should be the place where we are running this function from.

Reply:
Is the ``universal_newlines=True,`` that fixes it

Reply:
IMO, this is a patch affecting only `py38` (which is not officially supported for v1.21.x), it let us pass the testsuite without modifying our CI. We may investigate it to find the difference between this branch and `develop`, but to be honest, we have come across this because of this patch release, otherwise we would have kept working without noticing it, and this change won't affect `develop`...